### DISCUSSION TITLE
Accu-Chek Instant Glucose Meter Connectivity

### DISCUSSION THREAD

**Discussion by akerge** [NightscoutFoundation/xDrip#2997] [General]
Is the Instant glucose meter supported? 

Been trying to connect xdrip+ (`23ccbd1.2023.07.16`) with the Instant but it fails to upload any readings  although it is paired. Was using the same meter with mySugr successfully earlier. And did unpair before re-pairing with xdrip. Also, it seems not to be able to turn on the "Use Bluetooth Meter" in the "Glucose Meters" settings section.

If it's unsupported and a meter is necessary would be happy to provide one.

**Comment by ojalaj** [comment]
I'm experiencing exactly the same issue. I paired the meter through the phone BT settings, but it seems to be listed as Paired in xDrip+ list of glucose meters. When connecting, first is says that there is some small time difference between the devices and then it gives: "Timed out on: notify new glucose context."

xDrip+ experts - any thoughts?

**Reply by ojalaj** [reply]
@akerge Did you ever find a solution for this?

**Reply by akerge** [reply]
> @akerge Did you ever find a solution for this?

No, I went and bought the Contour. There's no support at the moment. Fastest way to get it is to write it.

**Comment by HammyHavoc** [comment]
If nobody else is working on this, I may have a go.

**Reply by Navid200** [reply]
I don't believe any one is working on it.
Please go ahead.

**Reply by ojalaj** [reply]
@HammyHavoc this would be great, if you have will and a way to do it!

**Comment by yhonatanb20** [comment]
Hey there all. Just experiencing the exact same issue with the instant, latest nightly build. Have there been any news on this?

**Reply by ojalaj** [reply]
No, nothing that I'm aware of. Maybe @HammyHavoc might have some insight into this?

**Comment by NiklasMehner** [comment]
https://github.com/NightscoutFoundation/xDrip/pull/4000/files should fix this (got a nice PR number for this one :)

As far as I understand the spec https://www.bluetooth.com/wp-content/uploads/Files/Specification/HTML/GLS_v1.0.1/out/en/index-en.html#UUID-45a50e27-d452-3609-08f4-8204fa4b87ae the "Glucose Measurement Context" is optional and ACCU-CHEK Instant does not provide it. This causes XDrip to wait for an answer of an non-existing context and time out.

With this change I can get measurements, but I have no other meters to test if this might break others (but it should not). And perhaps someone with deeper knowledge of the code can say if the data from the context is actually needed (handled in "processContextData").

**Comment by Navid200** [comment]
Thanks to @NiklasMehner, this should be working as of the 2025.05.23 Nightly release.

**Reply by mascheck55** [reply]
does it work with accu chek instant as the one and only unit connected? I have no other device and it will connect but does not show any data.

**Reply by vpv1952** [reply]
anybody has this accu-chek instant working with xDrip+?  I get it um pair, but on data is received.

**Comment by vpv1952** [comment]
recently I started to use the accu-chek instant meter, connected to xdrip.  it works: I take a finger prick blood measurement and the meter sends it to xdrip for calibration.  however, i've noticed something strange.  there is an error msg in the log that repeats over and over.  the photos show the msgs:

![Screenshot_2025-12-09-14-10-06-821_com eveningoutpost dexdrip](https://github.com/user-attachments/assets/38110c3f-fa46-4641-9753-26f087ae3fc5)
![Screenshot_2025-12-09-14-10-16-386_com eveningoutpost dexdrip](https://github.com/user-attachments/assets/168511eb-3df5-43f1-b269-9702e491fe52)
![Screenshot_2025-12-09-14-12-04-668_com eveningoutpost dexdrip](https://github.com/user-attachments/assets/abdff0d4-c572-4977-9a63-5659e1b5c3b9)

**Reply by vpv1952** [reply]
the msg is repeated for each day. and this stops on november 7.  why is this?
